faq.md

```diff
@@ -1,45 +1,634 @@
+
+```c#
+public class NativeWindowSettings
+{
+    public string Title { get; set; }              // Text for the title bar
+    public WindowIcon Icon { get; set; }           // Array of images of various sizes
+    
+    public bool StartFocused { get; set; }         // True by default
+    public bool StartVisible { get; set; }         // True by default
+    
+    public Vector2i? Location { get; set; }        // Top-left corner position
+    public Vector2i? Size { get; set; }            // Client/content size, in pixels
+    
+    public WindowBorder WindowBorder { get; set; } // Fixed, Resizable, Hidden (no bord.)
+    public WindowState WindowState { get; set; }   // Normal, Minim., Maxim., Hidden
+    public bool IsFullscreen { get; set; }         // Like Maximized, only bigger :-)
+}
+```
+
+A number of these, like `Location` and `Size` and `WindowState` and `IsFullscreen`, are re-exposed after the window is constructed as properties on the window itself, so you can use the equivalent window properties to update your window’s appearance and state as well.
+
+If you want to delay your window’s position/layout until after it has been constructed and important data for it has been loaded (like its position), a common pattern is something like this:
+
+```c#
+var nativeSettings = new NativeWindowSettings { StartVisible = false };
+using (var myWindow = new MyWindow(nativeSettings, gameSettings))
+{
+    myWindow.Run();
+}
+
+...
+    
+public class MyWindow : GameWindow
+{
+    ...
+        
+    public override OnLoad()
+    {
+        // Do setup...
+        ...
+            
+        // Now show it.
+        CenterWindow();
+        WindowState = WindowState.Normal;
+    }
+}
+```
+
+(Note that OpenTK 3 used a very different technique for setting up its initial window state:  Window parameters are passed as part of the `GameWindow` or `NativeWindow` constructor; there is no such thing as `NativeWindowSettings`; and configuration options are fairly limited.)
+
+### What’s up with OpenTK’s matrices?
+
+For historical reasons, OpenTK’s built-in `Matrix` structs store their data in row-major order, not column-major order.  This provides consistent backward-compatibility with existing software that uses OpenTK, but it can be confusing, for two reasons:
+
+- First, if you copy other code found online for OpenGL, you may need to *transpose* the matrices their code uses to get the expected behavior.
+- Second, matrix-matrix multiplications typically need to be done in _reverse_ order of those of most other libraries.
+
+This is a common source of confusion when using OpenTK, since it often prevents “code found on the Internet” from working directly.  We’re sorry if it causes you trouble, but it’s easy to work around it, and because so much existing software depends on it, it’s not something we plan to change any time soon.
```

Comment:
I think we need practical guide here too about how you get consistent multiplication order etc, but I can write that at a later time. It doesn't need to be part of this PR.

Reply:
Agreed.  We do need a good guide on that, describing all the different ways different things handle matrixes, but that's outside the scope of this.  If you write it up, we can add a link to it.